share/rocm/cmake/ROCMCreatePackage.cmake

```diff
@@ -222,50 +272,44 @@ macro(rocm_create_package)
     endif()
     if(PARSE_COMPONENTS)
         rocm_set_comp_cpackvar(PARSE_HEADER_ONLY "${PARSE_COMPONENTS}")
+        if(ROCM_PACKAGE_COMPONENT_DEPENDENCIES)
+            foreach(COMP_DEP IN LISTS ROCM_PACKAGE_COMPONENT_DEPENDENCIES)
+                string(REGEX REPLACE "^(.*)->.*$" "\\1" _downstream "${COMP_DEP}")
+                string(REGEX REPLACE "^.*->(.*)$" "\\1" _upstream "${COMP_DEP}")
+                string(TOUPPER "${_upstream}" _upstream_uc)
+                if(DEFINED "CPACK_DEBIAN_${_upstream_uc}_PACKAGE_NAME")
+                    set(_upstream_deb_package "${CPACK_DEBIAN_${_upstream_uc}_PACKAGE_NAME}")
+                elseif(DEFINED CPACK_DEBIAN_PACKAGE_NAME)
+                    string(TOLOWER "${CPACK_DEBIAN_PACKAGE_NAME}-${_upstream}" _upstream_deb_package)
+                else()
+                    string(TOLOWER "${CPACK_PACKAGE_NAME}-${_upstream}" _upstream_deb_package)
+                endif()
+                if(DEFINED "CPACK_RPM_${_upstream_uc}_PACKAGE_NAME")
+                    set(_upstream_rpm_package "${CPACK_RPM_${_upstream_uc}_PACKAGE_NAME}")
+                elseif(DEFINED CPACK_RPM_PACKAGE_NAME)
+                    string(TOLOWER "${CPACK_RPM_PACKAGE_NAME}-${_upstream}" _upstream_rpm_package)
+                else()
+                    string(TOLOWER "${CPACK_PACKAGE_NAME}-${_upstream}" _upstream_rpm_package)
+                endif()
+                rocm_package_add_rpm_dependencies(COMPONENT "${_downstream}" DEPENDS "${_upstream_rpm_package} >= ${CPACK_PACKAGE_VERSION}")
+                rocm_package_add_deb_dependencies(COMPONENT "${_downstream}" DEPENDS "${_upstream_deb_package} >= ${CPACK_PACKAGE_VERSION}")
+            endforeach()
+        endif()
     endif()
     include(CPack)
+    set(ROCM_PACKAGE_CREATED TRUE CACHE INTERNAL "Track whether rocm_create_package has been called.")
```

Comment:
I dont think a cached variable will work as this will be set to TRUE the second time cmake is ran. Instead. you could do `set(ROCM_PACKAGE_CREATED TRUE)` and then just check `if(DEFINED ROCM_PACKAGE_CREATED)` in the `rocm_package_add` functions.

Reply:
I used a cache variable to deal with the scoping constraints of CMake, because if we use `set(ROCM_PACKAGE_CREATED TRUE)` here then the variable will only be set in that scope. For example, in rocBLAS we perform a bunch of setup in the top level CMakeLists.txt, then include `cmake/packaging.cmake` to generate the package. Using the non-cache variable, if another component was set up in the top level CMakeLists.txt after including `cmake/packaging.cmake`, then that wouldn't be warned about. You are right though that a cache variable isn't ideal, as it does cause issues on a second run. I'll look into alternative solutions.

Reply:
You could use a global property.

Reply:
The solution I have currently implemented still uses a cache variable, but the variable is set to false at the top level of `ROCMCreatePackage.cmake`. This resets the variable to false on future builds, so there should be no spurious warnings. A global property would also work, but I think it would require slightly more boilerplate code.